.github/workflows/canary-integration-test.yml

```diff
@@ -1265,6 +1265,53 @@ jobs:
           # the check is not super ideal since 'mirroring_failed' is only displayed when there is a failure but not when it's working...
           timeout 60 sh -c 'while [ "$(kubectl exec -n rook-ceph deploy/rook-ceph-tools -t -- ceph fs snapshot mirror daemon status|jq -r '.[0].filesystems[0]'|grep -c "mirroring_failed")" -eq 1 ]; do echo "waiting for filesystem to be mirrored" && sleep 1; done'
 
+      - name: Create subvolume on primary cluster
+        run: |
+          kubectl exec -n rook-ceph deploy/rook-ceph-tools -t -- ceph fs subvolume create myfs testsubvolume
+
+      - name: Create subvolume of same name on secondary cluster
+        run: |
+          kubectl exec -n rook-ceph-secondary deploy/rook-ceph-tools -t -- ceph fs subvolume create myfs testsubvolume
+
+      - name: Deploy Direct Tools pod on primary cluster
+        run: |
+          kubectl -n rook-ceph create -f deploy/examples/direct-mount.yaml
+
+      - name: Deploy Direct Tools pod on secondary cluster
+        run: |
+          sed -i "s/rook-ceph # namespace/rook-ceph-secondary # namespace/" deploy/examples/direct-mount.yaml
+          kubectl -n rook-ceph-secondary create -f deploy/examples/direct-mount.yaml
+
+      - name: Configure a directory for snapshot mirroring on primary cluster
+        run: |
+          kubectl exec -n rook-ceph deploy/rook-ceph-tools -t -- ceph fs snapshot mirror enable myfs
+          kubectl exec -n rook-ceph deploy/rook-ceph-tools -t -- ceph fs snapshot mirror add myfs /volumes/_nogroup/testsubvolume/
+
+      - name: make sure that snapshot mirror is enabled on the secondary cluster
+        run: |
+          kubectl exec -n rook-ceph-secondary deploy/rook-ceph-tools -t -- ceph fs snapshot mirror enable myfs
+
+      - name: Create 3 snapshots on cluster primary cluster
+        run: |
+          kubectl exec -n rook-ceph deploy/rook-ceph-tools -t -- ceph fs subvolume snapshot create myfs testsubvolume snap1
+          kubectl exec -n rook-ceph deploy/rook-ceph-tools -t -- ceph fs subvolume snapshot create myfs testsubvolume snap2
+          kubectl exec -n rook-ceph deploy/rook-ceph-tools -t -- ceph fs subvolume snapshot create myfs testsubvolume snap3
+
+      - name: Get the peer and verify the peer synchronization status that snaps have synced on secondary cluster
+        run: |
+          mirror_daemon=$(kubectl -n rook-ceph exec deploy/rook-ceph-fs-mirror -- ls /var/run/ceph/|grep "fs-mirror"| head -n 1)
+          sleep 10s
```

Comment:
@gauravsitlani remove this sleep here and below instead wait for the resources to be ready, we try to avoid using sleep in ci as much as possible to have a faster CI

Reply:
sure